cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -57,7 +61,196 @@ type ClusterState struct {
+
+		// If the cluster version is lower than the newest patch version, we should inform about the possible upgrade
+		if patchUpgradePossible(clusterVersion, patchVersion) {
+
+			// The kubeadm version has to be upgraded to the latest patch version
+			newKubeadmVer := patchVersionStr
+			if kubeadmVersion.AtLeast(patchVersion) {
+				// In this case, the kubeadm CLI version is new enough. Don't display an update suggestion for kubeadm by making .NewKubeadmVersion equal .CurrentKubeadmVersion
+				newKubeadmVer = kubeadmVersionStr
+			}
+
+			upgrades = append(upgrades, Upgrade{
+				Description: description,
+				Before:      beforeState,
+				After: ClusterState{
+					KubeVersion:    patchVersionStr,
+					DNSVersion:     dns.GetKubeDNSVersion(patchVersion),
+					KubeadmVersion: newKubeadmVer,
+					// KubeletVersions is unset here as it is not used anywhere in .After
+				},
+			})
+		}
+	}
+
+	if canDoMinorUpgrade {
+		upgrades = append(upgrades, Upgrade{
+			Description: "stable version",
+			Before:      beforeState,
+			After: ClusterState{
+				KubeVersion:    stableVersionStr,
+				DNSVersion:     dns.GetKubeDNSVersion(stableVersion),
+				KubeadmVersion: stableVersionStr,
+				// KubeletVersions is unset here as it is not used anywhere in .After
+			},
+		})
+	}
+
+	if experimentalUpgradesAllowed || rcUpgradesAllowed {
+		// dl.k8s.io/release/latest.txt is ALWAYS an alpha.X version
+		// dl.k8s.io/release/latest-1.X.txt is first v1.X.0-alpha.0 -> v1.X.0-alpha.Y, then v1.X.0-beta.0 to v1.X.0-beta.Z, then v1.X.0-rc.1 to v1.X.0-rc.W.
+		// After the v1.X.0 release, latest-1.X.txt is always a beta.0 version. Let's say the latest stable version on the v1.7 branch is v1.7.3, then the
+		// latest-1.7 version is v1.7.4-beta.0
+
+		// Worth noticing is that when the release-1.X branch is cut; there are two versions tagged: v1.X.0-beta.0 AND v1.(X+1).alpha.0
+		// The v1.(X+1).alpha.0 is pretty much useless and should just be ignored, as more betas may be released that have more features than the initial v1.(X+1).alpha.0
+
+		// So what we do below is getting the latest overall version, always an v1.X.0-alpha.Y version. Then we get latest-1.(X-1) version. This version may be anything
+		// between v1.(X-1).0-beta.0 and v1.(X-1).Z-beta.0. At some point in time, latest-1.(X-1) will point to v1.(X-1).0-rc.1. Then we should show it.
+
+		// The flow looks like this (with time on the X axis):
+		// v1.8.0-alpha.1 -> v1.8.0-alpha.2 -> v1.8.0-alpha.3 | release-1.8 branch | v1.8.0-beta.0 -> v1.8.0-beta.1 -> v1.8.0-beta.2 -> v1.8.0-rc.1 -> v1.8.0 -> v1.8.1
+		//                                                                           v1.9.0-alpha.0                                             -> v1.9.0-alpha.1 -> v1.9.0-alpha.2
+
+		// Get and output the current latest unstable version
+		latestVersionStr, latestVersion, err := versionGetterImpl.VersionFromCILabel("latest", "experimental version")
```

Comment:
Nit: I wouldn't really call "latest" as "experimental". In case of `ci/*` builds I agree to call them experimental, but alphas/betas already have some quality tags attached (even were not very high).

Reply:
They are very experimental -- they are just cut from master without any special coordination that feature X, Y, Z must be there for it to ship.